src/components/button/templates/component/template.js

```diff
@@ -69,27 +51,106 @@ export function componentTemplate({ props } : { props : Object }) : string {
+            color: getButtonConfig(dual, 'dualColors')[color]
+        });
+    }
+
+    return buttons;
+}
 
-    let enableDualBtn = dual ? getButtonConfig(label, 'allowDualButton') : false;
-    enableDualBtn = !branding ? false : enableDualBtn;
+function buttonTemplate({ label, color, isDual, locale, branding } : { label : string, color : string, isDual : boolean, branding : boolean, locale : Object }) : string {
 
-    let logoColor   = getButtonConfig(label, 'logoColors')[color];
-    let taglineColor = getButtonConfig(label, 'tagLineColors')[color];
+    let logoColor = getButtonConfig(label, 'logoColors')[color];
 
-    let contentText = enableDualBtn ? getButtonConfig(label, 'defaultDualLabel') : (getButtonConfig(label, 'label') || content[label]);
+    let content = getLocaleContent(locale);
+
+    let contentText = isDual
+        ? getButtonConfig(label, 'dualLabel', content[label])
+        : getButtonConfig(label, 'label', content[label]);
 
     if (!branding) {
         contentText = removeBranding(contentText);
     }
 
-    let allowTagline = (tagline && branding && !fundingicons);
-    let tagContent   = enableDualBtn ? (content[getButtonConfig(label, 'defaultDualTagKey')] || '') : (content[getButtonConfig(label, 'tagkey')] || '');
+    contentText = expandContentText(contentText, { color, logoColor });
+
+    return `
+        <div class="paypal-button-content paypal-label-${ label } paypal-color-${ color } paypal-logo-color-${ logoColor }">
+            ${ contentText }
+        </div>
+    `;
+}
+
+function enableTagline({ tagline, branding, fundingicons }) : boolean {
+    return Boolean(tagline && branding && !fundingicons);
+}
+
+function taglineTemplate({ label, tagline, branding, fundingicons, color, dual, locale }) : string {
 
-    let labelText = expandContentText(contentText, { color, logoColor });
+    let enabled = enableTagline({ tagline, branding, fundingicons });
 
-    let secondButtonHtml = enableDualBtn ? getSecondBtnHtml(color) : '';
+    let key = dual
+        ? getButtonConfig(label, 'dualTagKey')
+        : getButtonConfig(label, 'tagKey');
+
+    let content = getLocaleContent(locale);
+    let text    = content[key];
+
+    let tagColor = getButtonConfig(label, 'tagLineColors')[color];
+
+    return `
+        <div class="paypal-button-tag-content paypal-tagline-color-${ tagColor }">
+            ${ (enabled && text) ? text : '' }
+        </div>
+    `;
+}
+
+export function componentTemplate({ props } : { props : Object }) : string {
+
+    let { label, locale, color, shape, branding,
+          fundingicons, tagline, dual } = normalizeProps(props);
+
+    let buttonHTML = determineButtons({ label, color, dual }).map(button => {
+        return buttonTemplate({
+            label: button.label,
+            color: button.color,
+            isDual: Boolean(dual),
+            locale,
+            branding
+        });
+    }).join('\n');
+
+    let taglineHTML = taglineTemplate({
```

Comment:
I added the check to the function itself, so we can contain all the logic there.